### PR TITLE
Remove Google+ dependency

### PR DESCRIPTION
**Pull request by Guisch**
Based on this: https://github.com/mstade/passport-google-oauth2/issues/13#issuecomment-459733198

I updated profile to add the new field and put them in the old ones to maximise compatibility

### CODE REVIEW

File: example/app.js [hunk 1]
```diff
@@ -100,11 +100,10 @@ app.get('/login', function(req, res){
 //   redirecting the user to google.com.  After authorization, Google
 //   will redirect the user back to this application at /auth/google/callback
 app.get('/auth/google', passport.authenticate('google', { scope: [
-       'https://www.googleapis.com/auth/plus.login',
-       'https://www.googleapis.com/auth/plus.profile.emails.read'] 
+       'email', 'profile'] 
 }));
 
-// GET /auth/google/callback
+// GET /auth/goofgle/callback
```
Comment (mstade) [inline example/app.js]: Typo?

```suggestion
// GET /auth/google/callback
```

File: lib/oauth2.js [hunk 2]
```diff
@@ -88,24 +88,35 @@ Strategy.prototype.authenticate = function(req, options) {
...
       var profile = { provider: 'google' };
-      profile.id           = json.id;
-      profile.displayName  = json.displayName;
-      profile.name         = json.name;
+      profile.id           = json.id || json.sub;
+      profile.displayName  = json.name;
+      profile.name         = {
+        givenName: json.given_name,
+        familyName: json.family_name
+      };
```
Comment (mstade) [inline lib/oauth2.js]: This (and the `displayName` change) make this an API breaking change. Not a big deal, but version bump should reflect this.

File: lib/oauth2.js [hunk 3]
```diff
@@ -88,24 +88,35 @@ Strategy.prototype.authenticate = function(req, options) {
...
       if (json.birthday) profile.birthday = json.birthday;
       if (json.relationshipStatus) profile.relationship = json.relationshipStatus;
       if (json.objectType && json.objectType == 'person') {
         profile.isPerson = true;
       }
       if (json.isPlusUser) profile.isPlusUser = json.isPlusUser;
+      if (json.email_verified !== undefined) {
+        profile.email_verified = json.email_verified;
+	profile.verified = json.email_verified;
```
Comment (mstade) [inline lib/oauth2.js]: Indentation is off.

```suggestion
	 profile.verified = json.email_verified;
```

File: lib/oauth2.js [hunk 4]
```diff
@@ -88,24 +88,35 @@ Strategy.prototype.authenticate = function(req, options) {
...
+      }
       if (json.placesLived) profile.placesLived = json.placesLived;
       if (json.language) profile.language = json.language;
+      if (!json.language && json.locale) {
+        profile.language = json.locale;
+	profile.locale = json.local;
```
Comment (mstade) [inline lib/oauth2.js]: Indentation is off.

```suggestion
	 profile.locale = json.local;
```

File: lib/oauth2.js [hunk 5]
```diff
@@ -115,15 +126,33 @@ Strategy.prototype.userProfile = function(accessToken, done) {
             return true
           }
         })
-      } 
+      }
+      if(!profile.email && json.email) {
```
Comment (mstade) [inline lib/oauth2.js]: Inconsistent spacing.

```suggestion
      if (!profile.email && json.email) {
```

File: lib/oauth2.js [hunk 6]
```diff
@@ -115,15 +126,33 @@ Strategy.prototype.userProfile = function(accessToken, done) {
...
+	profile.email = json.email;
```
Comment (mstade) [inline lib/oauth2.js]: Indentation is off.

```suggestion
	 profile.email = json.email;
```

File: lib/oauth2.js [hunk 7]
```diff
@@ -115,15 +126,33 @@ Strategy.prototype.userProfile = function(accessToken, done) {
...
+      }
+      if(!profile.emails && profile.email) {
+        profile.emails = [{
+	  value: profile.email,
+   	  type: "account"
```
Comment (mstade) [inline lib/oauth2.js]: Indentation is off.

```suggestion
   	   type: "account"
```

File: lib/oauth2.js [hunk 8]
```diff
@@ -115,15 +126,33 @@ Strategy.prototype.userProfile = function(accessToken, done) {
...
           }
         })
-      } 
+      }
+      if(!profile.email && json.email) {
+	profile.email = json.email;
+      }
+      if(!profile.emails && profile.email) {
+        profile.emails = [{
+	  value: profile.email,
```
Comment (mstade) [inline lib/oauth2.js]: Indentation is off.

```suggestion
	   value: profile.email,
```

File: lib/oauth2.js [hunk 9]
```diff
@@ -115,15 +126,33 @@ Strategy.prototype.userProfile = function(accessToken, done) {
             return true
           }
         })
-      } 
+      }
+      if(!profile.email && json.email) {
+	profile.email = json.email;
+      }
+      if(!profile.emails && profile.email) {
```
Comment (mstade) [inline lib/oauth2.js]: Inconsistent spacing.

```suggestion
      if (!profile.emails && profile.email) {
```

File: lib/oauth2.js (hunk 2 above)
Comment (Guisch) [reply lib/oauth2.js]: https://github.com/mstade/passport-google-oauth2/pull/15/commits/b94250c2d3be003db149eec672ae54897cc79186

**Comment by Guisch** [comment]
I tested on my server, work just fine.
I did not launch your tests tho since it needs to be modified as well and I do not know how

**Comment by jeremiepape** [comment]
Sounds great!
Could be ESlinted for a last check and cleanup before commit/push.
Thanks guys

**Comment by mstade** [comment]
Many thanks for this @Guisch – appreciate the assist! I checked and verified these changes as best I could locally, added some changes necessary to get the example working and then bumped to v0.2.0 before releasing to npm.

Hope this helps!

**Pull request merged**